File: 이재진/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        
        Set<Integer> w = new HashSet<>(Arrays.asList(nums));
        
        int answer = 0;
        int n = (int) nums.length/2;
        Map<Integer, Integer> m = new HashMap<>();
        for (int i=0; i<nums.length; i++) {
            if (m.keySet().contains(nums[i])) {
                m.put(nums[i], m.get(nums[i]+1));
            } else {
                m.put(nums[i], 1);
            }
        }
        System.out.println(m.keySet().size());
        if (m.keySet().size() >= n) {
            answer = n;
        } else {
            answer = m.keySet().size();
        }
        return answer;
    }
}

File: 정종길/프로그래머스/기타/폰켓몬.py
def solution(nums):
    num_pick = len(nums) // 2
    ls_types = list(set(nums))
    num_types = len(ls_types)
    # answer = 0
    if num_types <= num_pick:
        answer = num_types
    else:
        answer = num_pick
    return answer